server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -661,6 +661,14 @@ private void internalRecoverFromStore(IndexShard indexShard) throws IndexShardRe
                 indexShard.recoveryState().getIndex().setFileDetailsComplete();
             }
             indexShard.openEngineAndRecoverFromTranslog();
+            if (indexShard.shouldSeedRemoteStore()) {
+                indexShard.getThreadPool().executor(ThreadPool.Names.GENERIC).execute(() -> {
+                    logger.info("Attempting to seed Remote Store via local recovery for {}", indexShard.shardId());
+                    indexShard.refresh("remote store migration");
+                });
+                indexShard.waitForRemoteStoreSync();
```

Comment:
We will need to verify that remote store is synced after we return from here . This function doesn't throw an exception if seeding is not complete . I will take it as a follow up item though .